zerver/tests/test_auth_backends.py

```diff
@@ -1934,6 +1954,72 @@ def test_idp_initiated_signin_subdomain_implicit_invalid(self) -> None:
             "info",
         )])
 
+    def test_social_auth_saml_idp_org_membership_success(self) -> None:
+        idps_dict = copy.deepcopy(settings.SOCIAL_AUTH_SAML_ENABLED_IDPS)
+        idps_dict['test_idp']['attr_org_membership'] = 'member'
+        with self.settings(SOCIAL_AUTH_SAML_ENABLED_IDPS=idps_dict):
+            account_data_dict = self.get_account_data_dict(email=self.email, name=self.name)
+            result = self.social_auth_test(account_data_dict,
+                                           subdomain='zulip',
+                                           expect_choose_email_screen=False,
+                                           extra_attributes=dict(member=['zulip']))
+            data = load_subdomain_token(result)
+            self.assertEqual(data['email'], self.email)
+            self.assertEqual(data['full_name'], self.name)
+            self.assertEqual(data['subdomain'], 'zulip')
+            self.assertEqual(result.status_code, 302)
```

Comment:
This isn't for this PR, but I wonder if we should either make this block of 5 lines a function, `assert_successful_login_redirect`, for deduplication?

Reply:
Yup, I've been thinking about that, because the amount of copy-paste with this block is a bit absurd.